### PR TITLE
kubectl: allow to preselect interesting container in logs

### PR DESCRIPTION
**Pull request by mfojtik**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR introduces the `kubectl.kubernetes.io/default-logs-container` annotation that can be placed on a `Pod`. This annotation will cause `kubectl logs` to preselect the container specified as value for this annotation. This helps in case the Pod has multiple containers, but only one container is interesting for logs. In that case, users don't have to pick the container name manually, by using the `-c` option or `--all-containers`. 

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
Users might specify the `kubectl.kubernetes.io/default-logs-container` annotation in a Pod to preselect container for kubectl logs.
```

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: staging/src/k8s.io/kubectl/pkg/polymorphichelpers/logsforobject.go [hunk 1]
```diff
@@ -69,6 +73,16 @@ func logsForObjectWithClient(clientset corev1client.CoreV1Interface, object, opt
 		return ret, nil
 
 	case *corev1.Pod:
+		// in case the "kubectl.kubernetes.io/default-logs-container" annotation is present, we preset the opts.Containers to default to selected
+		// container. This gives users ability to preselect the most interesting container in pod.
+		if annotations := t.GetAnnotations(); annotations != nil && len(opts.Container) == 0 && len(annotations[defaultLogsContainerAnnotationName]) > 0 {
+			containerName := annotations[defaultLogsContainerAnnotationName]
+			if exists, _ := findContainerByName(t, containerName); exists != nil {
```
Comment (php-coder) [inline staging/src/k8s.io/kubectl/pkg/polymorphichelpers/logsforobject.go]: Should we exclude init and ephemeral containers or there are some use-cases for them too?

Comment (mfojtik) [reply staging/src/k8s.io/kubectl/pkg/polymorphichelpers/logsforobject.go]: yeah, i would leave it to pod creator to define what he think is the most important container logs he want to see.

**Pull request merged**